src/AWS.Deploy.CLI/AWSUtilities.cs

```diff
@@ -28,26 +27,35 @@ public class AWSUtilities : IAWSUtilities
         private readonly IDirectoryManager _directoryManager;
         private readonly IOptionSettingHandler _optionSettingHandler;
         private readonly IServiceProvider _serviceProvider;
+        private readonly ICredentialProfileStoreChainFactory _credentialChainFactory;
+        private readonly ISharedCredentialsFileFactory _sharedCredentialsFileFactory;
+        private readonly IFallbackCredentialsFactoryFactory _fallbackCredentialsFactory;
 
         public AWSUtilities(
             IServiceProvider serviceProvider,
             IToolInteractiveService toolInteractiveService,
             IConsoleUtilities consoleUtilities,
             IDirectoryManager directoryManager,
-            IOptionSettingHandler optionSettingHandler)
+            IOptionSettingHandler optionSettingHandler,
+            ICredentialProfileStoreChainFactory credentialChainFactory,
+            ISharedCredentialsFileFactory sharedCredentialsFileFactory,
+            IFallbackCredentialsFactoryFactory fallbackCredentialsFactory)
         {
             _serviceProvider = serviceProvider;
             _toolInteractiveService = toolInteractiveService;
             _consoleUtilities = consoleUtilities;
             _directoryManager = directoryManager;
             _optionSettingHandler = optionSettingHandler;
+            _credentialChainFactory = credentialChainFactory;
+            _sharedCredentialsFileFactory = sharedCredentialsFileFactory;
+            _fallbackCredentialsFactory = fallbackCredentialsFactory;
         }
 
-        public async Task<AWSCredentials> ResolveAWSCredentials(string? profileName)
+        public async Task<Tuple<AWSCredentials, string?>> ResolveAWSCredentials(string? profileName)
```

Comment:
Please add comments explaining the logic behind the region retrieval at every step, as in why some places we are using the CredentialsChainFactory, and in other placed just passing null ...

Reply:
added some more comments. open to more suggestions though to make it more clear